Hometask 4 films

kv67 opened the pull request.

kv67 committed:
Реализована загрузка списка фильмов с сайта https://www.themoviedb.org с помощью его API

kv67 committed:
Реализовано сохранение данных в RoomDatabase
Создана View Model, через которую осуществляется работа с данными.
Реализована работа с выбранным фильмом.

kv67 committed:
Изменен способ получения view model для фрагментов
Добавлена возможность очистки (удаления) данных из БД.
Добавлена возможность повторить загрузку данных при возникновении ошибки в ходе загрузки.
Добавлен pull to refresh.

kv67 committed:
Изменена работа с SwipeRefreshLayout (скрыт индикатор загрузки).

baidala commented on app/src/main/java/kve/ru/firstproject/model/FilmViewModel.kt:
```diff
@@ -0,0 +1,204 @@
+    }
+
+    private fun saveFilmsToDb(films: List<BestMovie?>) {
+        val task = Runnable {
```
посмотрите корутины для асинхронной работы

baidala reviewed (changes requested):
замечание по архитектуре -  когда используете паттерн single activity + fragments    активити у вас - это контейнер для фрагментов. она  может отвечать за меню/навигацию/тулбар/права  т.е. общие вещи для приложения.   и не нужно из активити манипулировать внутренностями фрагмента. это не ее зона ответственности. фрагмент сам отображает/перерисовывает данные на екране. нужно вынести в фрагменты методы добавления/удаления в избранное.  и открывать детали фильма

kv67 committed:
Созданы классы-репозитории для работы с БД и выполнения запросов к удаленному серверу.
Методы, обрабатывающие действия пользователя, вынесены из MainActivity во фрагменты.
Для следующего ДЗ:
 - настроено взаимодействие с Firebase
 - реализовано получение настроек из Firebase RemoteConfig
 - создан сервис для обработки сообщений от Firebase CloudMessaging
